vector/src/main/java/im/vector/app/features/location/UrlMapProvider.kt

```diff
@@ -19,19 +19,31 @@ package im.vector.app.features.location
 import im.vector.app.BuildConfig
 import im.vector.app.core.resources.LocaleProvider
 import im.vector.app.core.resources.isRTL
+import im.vector.app.features.raw.wellknown.getElementWellknown
+import org.matrix.android.sdk.api.extensions.tryOrNull
+import org.matrix.android.sdk.api.raw.RawService
+import org.matrix.android.sdk.api.session.Session
 import javax.inject.Inject
 
 class UrlMapProvider @Inject constructor(
-        private val localeProvider: LocaleProvider
+        private val localeProvider: LocaleProvider,
+        private val session: Session,
+        private val rawService: RawService
 ) {
     private val keyParam = "?key=${BuildConfig.mapTilerKey}"
 
-    // This is static so no need for a fun
-    val mapUrl = buildString {
+    private val fallbackMapUrl = buildString {
         append(MAP_BASE_URL)
         append(keyParam)
     }
 
+    suspend fun getMapUrl(): String {
+        val upstreamMapUrl = tryOrNull { rawService.getElementWellknown(session.sessionParams) }
+                ?.mapTileServerConfig
+                ?.mapStyleUrl
+        return upstreamMapUrl ?: fallbackMapUrl
+    }
+
     fun buildStaticMapUrl(locationData: LocationData,
```

Comment:
I think this fun should also be impacted by the url set up in the .well-known file, since AFAIU, it can contain another API key.